xmppserver/src/main/java/org/jivesoftware/openfire/ldap/LdapGroupProvider.java

```diff
@@ -124,7 +133,67 @@ private Group getGroupByDN(LdapName groupDN, Set<String> membersToIgnore) throws
             ctx = manager.getContext(baseDN);
             Attributes attrs = ctx.getAttributes(relativeDN, standardAttributes);
 
-            return processGroup(ctx, attrs, membersToIgnore);
+            boolean paging = false;
+            int countPage=0;
+
+            NamingEnumeration<? extends Attribute> i = attrs.getAll();
+            String groupMemberField = manager.getGroupMemberField();
+            String searchString = groupMemberField.contains(";range=")?groupMemberField.substring(0,groupMemberField.indexOf(";range=")):groupMemberField;
+            while (i.hasMore())
+            {
+                Attribute attribute = i.next();
+                String key = attribute.getID().toLowerCase();
+                if (key.startsWith(searchString))
+                {
+                    paging=true;
+                    break;
+                }
+            }
+
+            if (!paging)
+            {
+                return processGroup(ctx, attrs, membersToIgnore);
+            }
+            else
+            {
+                //retrieve first 0-1499
```

Comment:
The size of the page should probably be configurable.

Reply:
it depends on the attribute that came back from ldap... AD automatically added the range 0-1499 ... and so on... so i think it was not configureable...

that is also the fundamental problem with such big groups... the range was added to the attribute by AD (ldap) so the OF paging fails.. because OF could not find the member attribute...

Reply:
ok i will take the value from ldap.pagedResultsSize